### PR TITLE
chore: added and improved TSDoc of pricing module

### PR DESCRIPTION
**Pull request by shahednasser**
- General fixes to typedoc plugins
- Added GitHub action to generate the reference
- Added and improved TSDocs related to the pricing module

### CODE REVIEW

**Comment by shahednasser** [comment]
@riqwan can you review the TSDocs under `packages/types/src/pricing`?

@carlos-r-l-rodrigues might also be good to have you review the TSDocs in `packages/types/src/shared-context.ts`

File: packages/types/src/shared-context.ts [hunk 1]
```diff
@@ -5,6 +5,18 @@ export type SharedContext = {
   manager?: EntityManager
 }
 
+/**
+ * @internal The interface tag is used to ensure that the type is documented similar to interfaces.
+ * @interface
+ * 
+ * A shared context object that is used to share resources between modules.
```
Comment (carlos-r-l-rodrigues) [inline packages/types/src/shared-context.ts]:
```suggestion
 * A shared context object that is used to share resources between the application and the module.
```

**Review by carlos-r-l-rodrigues — APPROVED** [review]
It looks great @shahednasser .
I left just a small suggestion.

File: packages/types/src/pricing/common/price-rule.ts [hunk 2]
```diff
@@ -26,6 +57,21 @@ export interface CreatePriceRuleDTO {
...
+/**
+ * 
+ * @interface
+ * 
+ * An object used to specify the necessary data to update a price rule.
+ * 
+ * @prop id - A string indicating the ID of the price rule to update.
+ * @prop price_set_id - A string indicating the ID of the associated price set.
+ * @prop rule_type_id - A string indicating the ID of the associated rule type.
+ * @prop is_dynamic - A boolean indicating whether the price rule is dynamic.
```
Comment (riqwan) [inline packages/types/src/pricing/common/price-rule.ts]: This isn't implemented yet, should we maybe add a note? or remove entirely from TS

Comment (shahednasser) [reply packages/types/src/pricing/common/price-rule.ts]: Sure, I've removed the prop from the TSDocs and added the `@ignore` TSDoc directly on it with a private note that it's not implemented yet. Let me know about other properties/methods/etc... that we should remove from the generated docs too.

Comment (riqwan) [reply packages/types/src/pricing/common/price-rule.ts]: Rest look great! :)

**Review by riqwan — APPROVED** [review]
LGTM! 🔥

**Pull request merged**